### PR TITLE
Fix HiDPI rendering issues

### PR DESCRIPTION
**Pull request by sevoku**
This PR includes some workarounds for broken HiDPI support in Gtk.

6cd6dea: fixes QuickTask icon blurriness on Mac
8998278: fixes MiniMap drawing on Windows

Both are *workarounds* and don't fix the actual issues inside Gtk. Related bugs: https://bugzilla.xamarin.com/show_bug.cgi?id=51376, https://bugzilla.xamarin.com/show_bug.cgi?id=51378

### CODE REVIEW

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.QuickTasks/QuickTaskMiniMapMode.cs [hunk 1]
```diff
@@ -247,9 +248,10 @@ public BgBufferUpdate (Minimpap mode)
 					
 					cr = Gdk.CairoHelper.Create (mode.backgroundBuffer);
 					
-					cr.LineWidth = 1;
-					int w = mode.backgroundBuffer.ClipRegion.Clipbox.Width;
-					int h = mode.backgroundBuffer.ClipRegion.Clipbox.Height;
+					cr.LineWidth = 1;
+					var displayScale = Platform.IsWindows ? GtkWorkarounds.GetScaleFactor (mode) : 1.0;
```
Comment (slluis) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.QuickTasks/QuickTaskMiniMapMode.cs]: displayScale is unused?

Comment (sevoku) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.QuickTasks/QuickTaskMiniMapMode.cs]: Oh, hhm, there was a reason why I removed it... I'll doublecheck. Its long ago.

Comment (sevoku) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.QuickTasks/QuickTaskMiniMapMode.cs]: Yeah, its useless. I tried to fix rendering also for Mac, but it didn't work and I missed to remove all remaining parts of it. Should be fine now with e2645b7.

Comment (sevoku) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.QuickTasks/QuickTaskMiniMapMode.cs]: So the only way I see to fix this on Mac is to patch Gtk, as proposed in https://bugzilla.xamarin.com/show_bug.cgi?id=51376

**Pull request merged**